test/suites/ubuntu-core-edgex-image/build.sh

```diff
@@ -1,47 +1,58 @@
 #!/bin/bash -ex
 
+# Remove the pc-gadget directory if it already exists
 rm -rf pc-gadget
 git clone https://github.com/snapcore/pc-gadget.git --branch=22
 
-# build gadget snap
+# Build gadget snap
 cd pc-gadget
-# extend the size of disk partitions to have sufficient capacity for EdgeX snaps
-yq e '(.volumes.pc.structure[] | select(.name=="ubuntu-seed") | .size)="1500M"' gadget.yaml -i
 
-# setup default options for snaps
+# Extend the size of disk partitions to have sufficient capacity for EdgeX snaps
+yq e -i '(.volumes.pc.structure[] | select(.name=="ubuntu-seed") | .size)="1500M"' gadget.yaml
+
+# Set up default options for snaps
 # AZGf0KNnh8aqdkbGATNuRuxnt1GNRKkV (edgexfoundry snap)
 # AmKuVTOfsN0uEKsyJG34M8CaMfnIqxc0 (edgex-device-virtual snap) 
-yq e '.defaults += {
+yq e -i '.defaults += {
   "AZGf0KNnh8aqdkbGATNuRuxnt1GNRKkV": {
     "app-options": true,
-    "security": false
+    "security-secret-store": "off"
```

Comment:
"security-secret-store": "off" is set here

Reply:
The yaml parser converts off to boolean false. You need to make sure the quotations are around the value as `"off"` inside the yaml file. Maybe you can achieve that by passing escaped double quotations in the value: `"\"off\""`

Reply:
It appears `security-secret-store: "off"` is also considered invalid:
```
Run configure hook of "edgexfoundry" snap if present                           /[   35.131563] edgexfoundry.configure[2686]: Defer startup: starting disabled services
Run configure hook of "edgexfoundry" snap if present                           -[   35.135486] edgexfoundry.configure[2686]: security service config error: invalid setting for security-secret-store: "off"
```

Reply:
I guess that the additional quotation marks`'` around `"off"` automatically inserted into gadget.yaml may be the cause of the issue

Reply:
yes, ` security-secret-store: '"off"'` is definitely wrong. It has to be: ` security-secret-store: "off"`

Reply:
right, but I'm having trouble finding a way to use yq to add 'off' to the YAML. The results I'm getting are either `'"off"'` or just `off`